fabric-client-tags-api-v1/src/client/java/net/fabricmc/fabric/api/tag/client/v1/ClientTags.java

```diff
@@ -63,7 +64,7 @@ public static Set<Identifier> getOrCreateLocalTag(TagKey<?> tagKey) {
 
 		if (ids == null) {
 			ids = ClientTagsLoader.loadTag(tagKey);
-			LOCAL_TAG_CACHE.put(tagKey, ids);
+			LOCAL_TAG_CACHE.put(tagKey, Collections.unmodifiableSet(ids));
 		}
 
 		return ids;
```

Comment:
This can still be modifiable when the cache isnt hit the first time around. I would move the  `Collections.unmodifiableSet(` to the return of ClientTagsLoader.loadTag

Reply:
Done